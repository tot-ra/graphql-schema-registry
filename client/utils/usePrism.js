import { useRef, useLayoutEffect } from 'react';
import Prism from 'prismjs';

function usePrism(code) {
	const ref = useRef();

	useLayoutEffect(() => {
		if (code && ref.current) {
			Prism.highlightElement(ref.current);
		}
	}, [code]);

	return code ? (
		<pre>
			<code ref={ref} className="language-graphql">
				{code.trim()}
			</code>
		</pre>
	) : null;
}

export default usePrism;
