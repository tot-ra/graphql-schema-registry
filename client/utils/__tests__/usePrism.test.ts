import { renderHook } from '@testing-library/react-hooks';
import usePrism from '../usePrism';

describe('tests for usePrism', () => {
	it('should return null when no code is provided', () => {
		const { result } = renderHook(() => usePrism());

		expect(result.current).toBeNull();
	});

	it('should return a JSX element when code is provided', () => {
		const { result } = renderHook(() => usePrism('test code'));

		expect(result.current).not.toBeNull();
		expect(result.current).toMatchSnapshot();
	});
});
