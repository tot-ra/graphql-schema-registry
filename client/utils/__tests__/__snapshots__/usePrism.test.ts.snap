// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for usePrism should return a JSX element when code is provided 1`] = `
<pre>
  <code
    className="language-graphql"
  >
    test code
  </code>
</pre>
`;
